src/test/java/org/sagebionetworks/bridge/sdk/integration/ExternalIdsTest.java

```
@@ -0,0 +1,86 @@
+package org.sagebionetworks.bridge.sdk.integration;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.sagebionetworks.bridge.sdk.DeveloperClient;
+import org.sagebionetworks.bridge.sdk.Roles;
+import org.sagebionetworks.bridge.sdk.integration.TestUserHelper.TestUser;
+import org.sagebionetworks.bridge.sdk.models.PagedResourceList;
+import org.sagebionetworks.bridge.sdk.models.users.ExternalIdentifier;
+
+import com.google.common.collect.Lists;
+
+public class ExternalIdsTest {
+
+    private static final int LIST_SIZE = 10;
+    private static final int PAGE_SIZE = (LIST_SIZE/2);
+    private static TestUser developer;
+    
+    @Before
+    public void before() {
+        developer = TestUserHelper.createAndSignInUser(ExternalIdsTest.class, false, Roles.DEVELOPER);
+    }
+    
+    @After
+    public void after() {
+        developer.signOutAndDeleteUser();
+    }
+    
+    @Test
+    public void canCRUDExternalIds() {
+        // In order to prevent this test conflicting with other tests, each identifier is "namespaced" 
+        // with a random prefix that we'll use in all the queries.
+        String prefix = RandomStringUtils.randomAlphabetic(4);
+        
+        List<String> identifiers = Lists.newArrayListWithCapacity(LIST_SIZE);
+        for (int i=0; i < LIST_SIZE; i++) {
+            identifiers.add(prefix+RandomStringUtils.randomAlphabetic(10));
+        }
+        DeveloperClient client = developer.getSession().getDeveloperClient();
+        try {
+            client.addExternalIds(identifiers);
```

Minor: We should move the addExternalIds() call before the try. Otherwise, if addExternalIds() fail, then deleteExternalIds will also fail, and that secondary failure will mask the original failure.